pages/api/dpu/setup.ts

```diff
@@ -10,17 +12,28 @@ const setupHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 		return;
 	}
 	try {
-		await removePreviousInstallations(jsonBody.installationId)	
+		await removePreviousInstallations(jsonBody.installationId)
 	} catch (err) {
 		console.error(`[setupHandler] Unable to remove previous installations for ${jsonBody.installationId}`, err);
-		res.status(500).json({"error": "Internal Server Error"});
+		res.status(500).json({ "error": "Internal Server Error" });
+		return;
+	}
+	// get user_id for the given install_id
+	const userId = await getUserIdByTopicName(jsonBody.installationId).catch((error: any) => {
+		console.error("[setupHandler/getUserIdByTopicName] Failed to fetch userId from the database.", error);
+		res.status(500).json({ "error": "Internal Server Error" });
```

Comment:
You are anyway returning a 404 status when the userId is null. 
This will do nothing. It is anyway inside the concurrent anonymous function.